dhcpv4/ztpv4/ztp.go

```diff
@@ -15,39 +17,60 @@ type VendorData struct {
 
 var errVendorOptionMalformed = errors.New("malformed vendor option")
 
-// ParseVendorData will try to parse dhcp4 options looking for more
-// specific vendor data (like model, serial number, etc).
-func ParseVendorData(packet *dhcpv4.DHCPv4) (*VendorData, error) {
+func parseV4VIVC(vd *VendorData, packet *dhcpv4.DHCPv4) error {
+	vivc := packet.VIVC()
+	for _, id := range vivc {
+		if id.EntID == iana.EntIDCiscoSystems {
+			vd.VendorName = "Cisco Systems"
+			//SN:0;PID:R-IOSXRV9000-CC
+			for _, f := range bytes.Split(id.Data, []byte(";")) {
+				p := bytes.Split(f, []byte(":"))
+				if len(p) != 2 {
+					return errVendorOptionMalformed
+				}
+
+				switch string(p[0]) {
+				case "SN":
+					vd.Serial = string(p[1])
+				case "PID":
+					vd.Model = string(p[1])
+				}
+			}
+		}
+	}
+	return nil
+}
+
+func parseV4VendorClass(vd *VendorData, packet *dhcpv4.DHCPv4) error {
```

Comment:
Same, I think the `V4` in the name is redundant, it is already in the package name.